packages/smooth_app/lib/pages/product_list_user_dialog_helper.dart

```diff
@@ -0,0 +1,221 @@
+                  ),
+                ),
+          actions: <Widget>[
+            TextButton(
+              onPressed: () => Navigator.pop(context),
+              child: Text(appLocalizations.cancel),
+            ),
+            ElevatedButton(
+              onPressed: () async {
+                final ProductList? productList = await showCreate(context);
+                if (productList != null) {
+                  all.clear();
+                  all.addAll(daoProductList.getUserLists());
+                  setState(() => addedLists = true);
+                }
+              },
+              child: const Text('new list'), // TODO(monsieurtanuki): localize
+            ),
+            ElevatedButton(
+              onPressed: () => Navigator.pop(context, true),
+              child: Text(appLocalizations.okay),
+            ),
+          ],
+        ),
+      ),
+    );
+    if (addedLists == false && result != true) {
+      return false;
+    }
+    final Set<String> possibleChanges = <String>{};
+    possibleChanges.addAll(withBarcode);
+    possibleChanges.addAll(newWithBarcode);
+    for (final String name in possibleChanges) {
+      if (withBarcode.contains(name) && newWithBarcode.contains(name)) {
+        continue;
+      }
+      if ((!withBarcode.contains(name)) && (!newWithBarcode.contains(name))) {
+        continue;
+      }
+      final ProductList productList = ProductList.user(name);
+      await daoProductList.set(
+          productList, barcode, newWithBarcode.contains(name));
+    }
+    return true;
+  }
+
+  Future<ProductList?> showRename(
+    final BuildContext context,
+    final ProductList initialProductList,
+  ) async {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    final TextEditingController _textEditingController =
+        TextEditingController();
+    final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+
+    final String initialName = initialProductList.parameters;
+    _textEditingController.text = initialName;
+    final String? newName = await showDialog<String>(
+      context: context,
+      builder: (final BuildContext context) => AlertDialog(
+        title: const Text('Rename user list'), // TODO(monsieurtanuki): localize
+        content: Form(
+          key: _formKey,
+          child: SmoothTextFormField(
+            type: TextFieldTypes.PLAIN_TEXT,
+            controller: _textEditingController,
+            hintText: appLocalizations
+                .username_or_email, // TODO(monsieurtanuki): localize
+            textInputAction: TextInputAction.done,
+            validator: (String? value) {
+              final List<String> lists = daoProductList.getUserLists();
+              if (value == null || value.isEmpty) {
+                return appLocalizations.login_page_username_or_email;
+              }
+              if (lists.contains(value) && value != initialName) {
+                return 'name already exists';
+              }
```

Comment:
Can probably be split into two different messages but a comment for that is fine